### PR TITLE
Fix first character lost when reconnect on ssh

### LINKED ISSUE #1455: First input char is lost after "okteto up" reconnects
**Describe the bug**
I am running `okteto up` and there is a reconnection even. For example, the pod is deleted. When my terminal is reconnected, the first input char is lost.

**Expected behavior**
No input char is lost

**Additional context**
We might need to inject a char in stdout when reconnecting.

### PR DESCRIPTION
**Pull request by jLopezbarb**
Signed-off-by: Javier López Barba <javier@okteto.com>

Fixes #1455

## Proposed changes
-  Checks if there is another process trying to copy from stdin to remote server and wait for it to finish to start copying.
-  Refactor copy to get the character lost
-  Copy that character into a channel and inject it into remote stdin

### CODE REVIEW

File: pkg/ssh/exec.go [hunk 1]
```diff
@@ -32,6 +31,9 @@ import (
 	"golang.org/x/crypto/ssh/terminal"
 )
 
+var lostCharacter = make(chan []byte, 1)
```
Comment (pchico83) [inline pkg/ssh/exec.go:34]: could we have a single channel?
if there are several reconnections, there could be several lost chars

File: pkg/ssh/exec.go [hunk 2]
```diff
@@ -184,6 +187,60 @@ More information is available here: https://okteto.com/docs/reference/manifest#r
 	return err
 }
 
+func waitForOtherCopyToFinish() {
+	wasRunning := false
+Loop:
```
Comment (pchico83) [inline pkg/ssh/exec.go:192]: is it possible to avoid this syntax and use a structured loop instead?

File: pkg/ssh/exec.go [hunk 3]
```diff
@@ -184,6 +187,60 @@ More information is available here: https://okteto.com/docs/reference/manifest#r
...
+	case char := <-lostCharacter:
+		remoteStdin.Write(char)
+	default:
+		break
+	}
+}
+
+func copyFromLocalToRemote(local io.Reader, remote io.WriteCloser, buf []byte) {
+	for {
+		nr, er := local.Read(buf)
```
Comment (pchico83) [inline pkg/ssh/exec.go:220]: could we do a `select`here between `Read` and the `lostCharacter` channel?

Comment (jLopezbarb) [reply pkg/ssh/exec.go:220]: no, I tried that but it will get blocked anyways on the read until you press a key

File: pkg/ssh/exec.go (hunk 2 above)
Comment (jLopezbarb) [reply pkg/ssh/exec.go:192]: yes, we could get out of the loop with a return statement and not use the tag

File: pkg/ssh/exec.go (hunk 1 above)
Comment (jLopezbarb) [reply pkg/ssh/exec.go:34]: I will try these user cases and try to have just one channel

**Pull request merged**